wremnants/include/muon_calibration.h

```diff
@@ -1235,6 +1231,83 @@ class ZNonClosureParametrizedHelperSplines {
     std::shared_ptr<const T> correctionHist_;
 };
 
+// the parametrized Z non-closure helper without the de-correlation in output nuisances
+// TODO: vectorize the helper for multiple muons
+template <typename T, size_t NEtaBins>
+class ZNonClosureParametrizedHelperSplinesCorl {
+
+public:
+    using hist_t = T;
+    using tensor_t = typename T::storage_type::value_type::tensor_t;
+    using out_tensor_t = Eigen::TensorFixedSize<double, Eigen::Sizes<2>>;
+
+    ZNonClosureParametrizedHelperSplinesCorl(T&& corrections):
```

Comment:
instead of a separate helper with lots of duplicated code, couldn't you just use the existing helper with 1 eta bin?

Reply:
Ok I guess this is needed for "realistic" non-closure uncertainties which need to be fully correlated anyways, so fine to keep it like this for the moment.

Reply:
This is indeed ugly, but the reason I did it this way is to avoid having an unnecessary eta axis for the correlated non-closure nuisance.
It was suggested to me that we get rid of extra dimensions in the histograms to make the output more comprehensible.
But I guess I can define 2 output types `Eigen::TensorFixedSize<double, Eigen::Sizes<2>>` and `Eigen::TensorFixedSize<double, Eigen::Sizes<NETABINS, 2>>` and pass in an extra flag for the call operator

Reply:
No you can only switch output types based on template parameters, and anyways extra runtime arguments to the call operator is bad for performance in principle.  Let's just keep it like this for now, since we may need this later for other reasons anyways.

Reply:
I see